src/utils.ts

```diff
@@ -100,6 +100,30 @@ function getPlatform(): string {
   return '' // unknown
 }
 
+/**
+ * Normalizes headers into a plain object format.
+ * This function handles various types of HeaderInit objects such as Headers, arrays of key-value pairs,
+ * and plain objects, converting them all into an object structure.
+ *
+ * @param {HeadersInit|undefined} headers - The headers to normalize. Can be one of the following:
+ *   - A `Headers` object from the Fetch API.
+ *   - A plain object with key-value pairs representing headers.
+ *   - An array of key-value pairs representing headers.
+ * @returns {Record<string,string>} - A plain object representing the normalized headers.
+ */
+function normalizeHeaders(headers?: HeadersInit | undefined): Record<string,string> {
+  if (headers instanceof Headers) {
+      // If headers are an instance of Headers, convert it to an object
+      return Object.fromEntries(headers.entries());
```

Comment:
```suggestion
      const obj: Record<string, string> = {};
        headers.forEach((value, key) => {
          obj[key] = value;
        });
        return obj;
```

Let's use `forEach` for compatibility reasons